### DISCUSSION TITLE
Octopart search

### DISCUSSION THREAD

**Discussion by pdo59** [Part-DB/Part-DB-server#186] [Ideas]
Hello,
it would be nice to integrate component search via the Octopart API into this excellent project. [https://octopart.com/business/api/v4/api-transition](url)
This way you could create a component much faster.

**Comment by jbtronics** [comment]
Something like this is planned, but it does not have the highest priority for me.

My idea is to build a system which can fetch data from multiple (configurable) data sources. The idea is to not be solely bound to Octopart (as it seems that they are restricting their free API more and more), but that you can use the APIs of other providers, like digikey (which provide attachments, parameters, and more of the components they sell).

This also has the advantage that in principle data providers for things which are not electronic parts, can be written.

**Comment by jbtronics** [comment]
See https://github.com/Part-DB/Part-DB-server/discussions/323
